### ISSUE TITLE
FALSE POSITIVE: RANGE_ARRAY_INDEX reports inside a contradictory null branch

### ISSUE THREAD

**Issue by rikki322** [spotbugs/spotbugs#4275]
Hi, I found a false positive in SpotBugs 4.10.3 when value cannot be both null and non-null, so the array access is unreachable.

**Affected tool**

`SpotBugs 4.10.3`

**Affected checker**

`SpotBugs RANGE_ARRAY_INDEX`

**Minimal reproducer**

```java
class SpotBugsUnreachableArrayIndex {
  void test(Object value) {
    if (value == null && value != null) {
      int[] array = new int[1];
      array[2] = 1;
    }
  }
}
```

**Reproduction command**

```bash
spotbugs -version
javac -d classes spotbugs-range-array-index-fp-unreachable.java
spotbugs -textui -effort:max -low -visitors DumbMethods classes
```

**Current behavior**

SpotBugs reports `RANGE_ARRAY_INDEX` at line 5.

**Expected behavior**

SpotBugs should not report `RANGE_ARRAY_INDEX` because value cannot be both null and non-null, so the array access is unreachable.
